### ISSUE TITLE
2.0.10: three test failures

### ISSUE THREAD

**Issue by 0-wiz-0** [rrthomas/libpaper#44]
When packaging 2.0.10 for pkgsrc, I saw three test failures.

Configured with:
```
./configure --sysconfdir=/usr/pkg/etc --enable-relocatable --prefix=/usr/pkg --build=x86_64--netbsd --host=x86_64--netbsd --mandir=/usr/pkg/man --enable-option-checking=yes
```

The errors all look like this:
```
@@ -1 +1 @@
-paper: bad height in line 1 of /scratch/print/libpaper/work/libpaper-2.0.10/tests/bad-height.10693/usr/pkg/etc/paperspecs
+paper: bad height in line 1 of /scratch/print/libpaper/work/libpaper-2.0.10/tests/bad-height.10693//usr/pkg/etc/paperspecs
FAIL bad-height.sh (exit status: 1)
```
i.e. there's a slash too many in the result that is not expected.
[test-suite.log](https://github.com/rrthomas/libpaper/files/11211458/test-suite.log)

**Comment by rrthomas** [comment]
Yeah, this has been reported before, and I have tried to work out what is going wrong in gnulib's relocatable code, but I couldn't. However, since it's not really a problem for paper, I will just adjust the test to be less fussy.

Thanks for packaging libpaper!

**Comment by 0-wiz-0** [comment]
Hm, I just tried that version but didn't see a change:
```
@@ -1 +1 @@
-paper: missing field in line 1 of /scratch/wip/libpaper2/work/libpaper-2.0.11/tests/paperspecs-invalid.9230/usr/pkg/etc/paperspecs
+paper: missing field in line 1 of /scratch/wip/libpaper2/work/libpaper-2.0.11/tests/paperspecs-invalid.9230//usr/pkg/etc/paperspecs
```

**Comment by rrthomas** [comment]
Thanks, I'll have another look.

**Comment by rrthomas** [comment]
I don't understand what is going wrong here. For one of the failing tests, please could you attach `$BASENAME-output.txt` and `$BASENAME-output-fixed.txt`?

**Comment by 0-wiz-0** [comment]
I'm sorry, but where are those files? I see
```
# ls -1  work/libpaper-2.0.11/tests/paperspecs-invalid.*
work/libpaper-2.0.11/tests/paperspecs-invalid.log
work/libpaper-2.0.11/tests/paperspecs-invalid.sh
work/libpaper-2.0.11/tests/paperspecs-invalid.trs
```
and the only `*.txt` files I see are the ones in `tests/*-expected.txt`.

**Comment by rrthomas** [comment]
Sorry, the test directories are all cleaned up automatically. I guess it shouldn't be in case of error. For now, you can just comment out the line starting `trap` towards the end of `tests/run-test`.

**Comment by 0-wiz-0** [comment]
[paperspecs-invalid-output-fixed.txt](https://github.com/rrthomas/libpaper/files/11213634/paperspecs-invalid-output-fixed.txt)
[paperspecs-invalid-output.txt](https://github.com/rrthomas/libpaper/files/11213635/paperspecs-invalid-output.txt)

**Comment by rrthomas** [comment]
Thanks very much. I'm baffled, when I run the sed script in `run-test` on the `paperspecs-invalid-output.txt`, it removes the double slash. Could FreeBSD sed be working differently?

**Comment by rrthomas** [comment]
I have pushed an alternative fix, to avoid having to diagnose either a regex or shell incompatibility/bug.

If you could possibly build libpaper from git and run the tests, I would be most grateful.

**Comment by 0-wiz-0** [comment]
I applied the diff manually to 2.0.11 and it fixed the problem, thanks!

I think you need `sed -E` if you want to use '+' in a portable way, but I'm not completely sure.

**Comment by rrthomas** [comment]
> I applied the diff manually to 2.0.11 and it fixed the problem, thanks!

Great, I'll make another release.

> I think you need `sed -E` if you want to use '+' in a portable way, but I'm not completely sure.

My understanding is that without `-E` sed should use basic POSIX regexs, and with `-E` it should use extended POSIX regexs, and GNU sed is documented as working that way, as well as the POSIX man page for sed. But in this case I'm quite happy to give up on understanding what's going on. (I was wondering if it's to do with shell evaluation of backslashes, but I'm using `/bin/sh` and it works for me, so…)

**Comment by rrthomas** [comment]
I've released 2.0.12.

**Comment by 0-wiz-0** [comment]
One more link then I'll shut up:
https://man.netbsd.org/re_format.7
sed(1) on NetBSD defaults to Basic regular expressions (if -E is not specified) which means according to that link, that '+' is an ordinary character.
Thanks for fixing the problem and making the release so quickly!
